src/core/meshcop/joiner.cpp

```diff
@@ -65,6 +67,19 @@ Joiner::Joiner(ThreadNetif &aNetif):
 ThreadError Joiner::Start(const char *aPSKd)
 {
     ThreadError error;
+    union
+    {
+        Mac::ExtAddress extAddress;
+        uint8_t buf[Crypto::Sha256::kHashSize];
+    };
+    Crypto::Sha256 sha256;
+
+    // use extended address based on factory-assigned IEEE EUI-64
+    otPlatRadioGetIeeeEui64(NULL, buf);
+    sha256.Start();
+    sha256.Update(buf, OT_EXT_ADDRESS_SIZE);
+    sha256.Finish(buf);
+    mNetif.GetMac().SetExtAddress(extAddress);
```

Comment:
Need to update link local address here:
`mNetif.GetMle().UpdateLinkLocalAddress();`